engine/consensus/sealing/counters/monotonous_counter.go

```diff
@@ -0,0 +1,36 @@
+package counters
+
+import "sync/atomic"
+
+// StrictMonotonousCounter is a helper struct which implements a strict monotonous counter.
+// StrictMonotonousCounter is implemented using atomic operations and doesn't allow to set a value
+// which is lower or equal to already stored. Uses atomic operations + spin lock to implement concurrent access guarantees
```

Comment:
```suggestion
// which is lower or equal to the already stored one. Uses atomic operations + spin lock to implement concurrent access guarantees
```

Reply:
But you are actually spinning in loop to make an update, pretty much the same as spinlock.